bazel/envoy_library.bzl

```diff
@@ -66,6 +66,62 @@ EXTENSION_SECURITY_POSTURES = [
     "data_plane_agnostic",
 ]
 
+# Extension categories as defined by factories
+EXTENSION_CATEGORIES = [
+    "envoy.access_logger.extension_filters",
+    "envoy.access_loggers",
+    "envoy.bootstrap",
+    "envoy.clusters",
+    "envoy.compression.compressor",
+    "envoy.compression.decompressor",
+    "envoy.dubbo_proxy.filters",
+    "envoy.dubbo_proxy.protocols",
+    "envoy.dubbo_proxy.route_matchers",
+    "envoy.dubbo_proxy.serializers",
+    "envoy.fatal_action",
+    "envoy.filters.http",
+    "envoy.filters.listener",
+    "envoy.filters.network",
+    "envoy.filters.udp_listener",
+    "envoy.filters.upstream_network",
+    "envoy.formatter",
+    "envoy.grpc_credentials",
+    "envoy.guarddog_actions",
+    "envoy.health_checkers",
+    "envoy.http.cache",
+    "envoy.internal_redirect_predicates",
+    "envoy.matching.action",
+    "envoy.matching.input_matcher",
+    "envoy.quic_client_codec",
+    "envoy.quic_server_codec",
+    "envoy.rate_limit_descriptors",
+    "envoy.request_id_extension",
+    "envoy.resolvers",
+    "envoy.resource_monitors",
+    "envoy.retry_host_predicates",
+    "envoy.retry_priorities",
+    "envoy.singleton",
+    "envoy.ssl_context_manager",
+    "envoy.stats_sinks",
+    "envoy.thrift_proxy.filters",
+    "envoy.thrift_proxy.protocols",
+    "envoy.thrift_proxy.transports",
+    "envoy.tls_handshakers",
+    "envoy.tls.key_providers",
+    "envoy.tracers",
+    "envoy.transport_sockets.downstream",
+    "envoy.transport_sockets.upstream",
+    "envoy.typed_metadata",
+    "envoy.udp_listeners",
+    "envoy.udp_packet_writers",
+    "envoy.upstream_options",
+    "envoy.upstreams",
+    "test",
+    "testing",
+    "testing.published",
+    "testing.published.additional.category",
```

Comment:
Can you shrink this down to those that are actually used? I don't see anything when I search for "additional.category" in this PR other than this line.

Reply:
yep, sure